src/Reflex/Dynamic.hs

```diff
@@ -232,12 +235,21 @@ distributeListOverDynPure =
     fromDSum :: DSum (Const2 Int a) Identity -> a
     fromDSum (Const2 _ :=> Identity v) = v
 
---TODO: Generalize this to functors other than Maps
 -- | Combine a 'Dynamic' of a 'Map' of 'Dynamic's into a 'Dynamic'
 -- with the current values of the 'Dynamic's in a map.
 joinDynThroughMap :: forall t k a. (Reflex t, Ord k) => Dynamic t (Map k (Dynamic t a)) -> Dynamic t (Map k a)
 joinDynThroughMap = joinDyn . fmap distributeMapOverDynPure
 
+-- | Combine a 'Dynamic' of a 'Traversable' of 'Dynamic's into a 'Dynamic'
+-- with the current values of the 'Dynamic's in the functor.
+joinDynThroughTraversable :: (Reflex t, Traversable f) => Dynamic t (f (Dynamic t a)) -> Dynamic t (f a)
```

Comment:
See https://github.com/alexfmpe/reflex/pull/1

Reply:
This looks like it will have bad performance characteristics.  `joinDynThroughMap` uses a single `merge` and ensures that changes are applied about as efficiently as possible to the output.  We've gotta do something like that here, otherwise we can't recommend that people do this generally.

Reply:
@ryantrinkle I don't think it gets better than `joinDynThroughTraversable = (>>= sequenceA)` with only the `Traversable` constraint. The problem is the cost of small updates to the potentially large container. But there's no way to do that better than O(n) using only `Traversable`. Even if you use something like `distributeListOverDynPure`, you still have to reify the new list into the original container type, which will be O(n).

Reply:
`sequenceA` ought to accumulate efficiently anyway unless the datatype is inherently biased against that (e.g. lists)